internal/resolution/pm/gomod/job.go

```diff
@@ -66,34 +76,149 @@ func (j *Job) Run() {
 	if err != nil {
-		j.Errors().Critical(util.NewPMJobError(err.Error()))
+		j.handleError(j.createError(err.Error(), cmd, status))
 	}
 }
 
-func (j *Job) runGraphCmd(workingDirectory string) ([]byte, error) {
+func (j *Job) runGraphCmd(workingDirectory string) ([]byte, string, error) {
 	graphCmd, err := j.cmdFactory.MakeGraphCmd(workingDirectory)
 	if err != nil {
-		return nil, err
+		return nil, graphCmd.String(), err
 	}
 
 	graphCmdOutput, err := graphCmd.Output()
 	if err != nil {
-		return nil, j.GetExitError(err, "")
+		return nil, graphCmd.String(), j.GetExitError(err, "")
 	}
 
-	return graphCmdOutput, nil
+	return graphCmdOutput, graphCmd.String(), nil
 }
 
-func (j *Job) runListCmd(workingDirectory string) ([]byte, error) {
+func (j *Job) runListCmd(workingDirectory string) ([]byte, string, error) {
 	listCmd, err := j.cmdFactory.MakeListCmd(workingDirectory)
 	if err != nil {
-		return nil, err
+		return nil, listCmd.String(), err
 	}
 
 	listCmdOutput, err := listCmd.Output()
 	if err != nil {
-		return nil, j.GetExitError(err, "")
+		return nil, listCmd.String(), j.GetExitError(err, "")
 	}
 
-	return listCmdOutput, nil
+	return listCmdOutput, listCmd.String(), nil
+}
+
+func (j *Job) createError(error string, cmd string, status string) job.IError {
+	cmdError := util.NewPMJobError(error)
+	cmdError.SetCommand(cmd)
+	cmdError.SetStatus(status)
+
+	return cmdError
+}
+
+func (j *Job) handleError(cmdError job.IError) {
+	expressions := []string{
+		versionNotFoundErrRegex,
+		dependencyNotFoundErrRegex,
+		noPackageErrRegex,
+		unableToResolveErrRegex,
+		noInternetErrRegex,
+	}
+
+	for _, expression := range expressions {
+		regex := regexp.MustCompile(expression)
+		matches := regex.FindAllStringSubmatch(cmdError.Error(), -1)
+
+		if len(matches) > 0 {
+			cmdError = j.addDocumentation(expression, matches, cmdError)
+			j.Errors().Append(cmdError)
+
+			return
+		}
+	}
+
+	j.Errors().Append(cmdError)
+}
+
+func (j *Job) addDocumentation(expr string, matches [][]string, cmdError job.IError) job.IError {
+	documentation := cmdError.Documentation()
+
+	switch expr {
+	case versionNotFoundErrRegex:
+		documentation = getVersionNotFoundErrorDocumentation(matches)
+	case dependencyNotFoundErrRegex:
+		documentation = getDependencyNotFoundErrorDocumentation(matches)
+	case noPackageErrRegex:
+		documentation = getNoPackageErrorDocumentation(matches)
+	case unableToResolveErrRegex:
+		documentation = getDependencyNotFoundErrorDocumentation(matches)
+	case noInternetErrRegex:
+		documentation = getNoInternetErrorDocumentation(matches)
+	}
+
+	cmdError.SetDocumentation(documentation)
+
+	return cmdError
+}
+
+func getVersionNotFoundErrorDocumentation(matches [][]string) string {
+	dependency := ""
+	recommendation := ""
+	if len(matches) > 0 && len(matches[0]) > 1 {
+		dependency = matches[0][1]
+		recommendation = matches[0][2]
+	}
+
+	return strings.Join(
+		[]string{
+			"Failed to find package",
+			dependency + ".",
```

Comment:
```suggestion
			"\"" + dependency + "\".",
```
I think its good to add quotation marks here as the other documentation does that around dependency names.